### DISCUSSION TITLE
请问如何配置内网穿透下的IP与端口？

### DISCUSSION THREAD

**Discussion by SeptSky** [duyanming/Anno.Core#11] [Announcements]
请问使用内网穿透的情况下，该如何配置Anno.config和Cocas.config这个文件中的IP地址和端口？

**Comment by duyanming** [comment]
1、注册中心：
Cocas.config 这个里面配置的是注册中心监听的地址；
```
 <!--#注册中心 配置-->
  <Port>6660</Port>
````
例如注册中心监听地址为：10.128.9.5:6660

2、业务服务：
Anno.config ：
```
   <!--业务服务监听端口-->
  <Port>6659</Port>

  <!--注册到的目标-->
  <Ts Ip="10.128.9.5" Port="6660"/>
```

只要：
 1、服务端根据设置的注册中心地址`  <Ts Ip="127.0.0.1" Port="6660"/>` 能访问到注册中心
 2、 注册中心能访问到业务服务端口就可以 (一般业务服务的IP都是自动获取的，不过也可以通过启动参数指定 -h )



3、参考：

```
启动参数：
	-p 6659		设置启动端口
	-xt 200		设置服务最大线程数
	-t 20000		设置超时时间（单位毫秒）
	-w 1		设置权重
	-h 192.168.0.2	设置服务在注册中心的地址
	-tr false		设置调用链追踪是否启用
```

**Reply by SeptSky** [reply]
目前已经验证的配置：Anno.Config使用127.0.0.1，内网穿透映射到该地址；Cocas.config配置为本机局域网Ip，如192.168.64.141，然后外网的客户端使用配置的公网域名和端口即可访问了。
感谢支持！
